Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py

```diff
@@ -432,41 +432,55 @@ def send_request(self, path: str, method: str = 'GET', body: dict = None, params
+
             try:
-                file_entry = file['id']
-                file_name = file['name']
-                shutil.copy(demisto.getFilePath(file_entry)['path'], file_name)
-                with open(file_name, 'rb') as f:
-                    res = requests.request(method, url, headers=headers, data=body, params=params, files={'file': f},
-                                           auth=(self._username, self._password), verify=self._verify,
-                                           proxies=self._proxies)
-                shutil.rmtree(demisto.getFilePath(file_entry)['name'], ignore_errors=True)
+                json_res = res.json()
             except Exception as err:
-                raise Exception('Failed to upload file - ' + str(err))
-        else:
-            res = requests.request(method, url, headers=headers, data=json.dumps(body) if body else {}, params=params,
-                                   auth=(self._username, self._password), verify=self._verify, proxies=self._proxies)
-
-        try:
-            json_res = res.json()
-        except Exception as err:
-            if not res.content:
-                return ''
-            raise Exception(f'Error parsing reply - {str(res.content)} - {str(err)}')
-
-        if 'error' in json_res:
-            message = json_res.get('error', {}).get('message')
-            details = json_res.get('error', {}).get('detail')
-            if message == 'No Record found':
-                return {'result': []}  # Return an empty results array
-            raise Exception(f'ServiceNow Error: {message}, details: {details}')
-
-        if res.status_code < 200 or res.status_code >= 300:
-            raise Exception(f'Got status code {str(res.status_code)} with url {url} with body {str(res.content)}'
-                            f' with headers {str(res.headers)}')
+                if not res.content:
+                    return ''
+                raise Exception(f'Error parsing reply - {str(res.content)} - {str(err)}')
+
+            if 'error' in json_res:
+                message = json_res.get('error', {}).get('message')
+                details = json_res.get('error', {}).get('detail')
+                if message == 'No Record found':
+                    return {'result': []}  # Return an empty results array
+                if res.status_code == 401:
+                    demisto.debug(f'Got status code 401 - {json_res}. Retrying ...')
+                else:
+                    raise Exception(f'ServiceNow Error: {message}, details: {details}')
+
+            if res.status_code < 200 or res.status_code >= 300:
+                if res.status_code != 401:
+                    raise Exception(
+                        f'Got status code {str(res.status_code)} with url {url} with body {str(res.content)}'
+                        f' with headers {str(res.headers)}')
+            else:
+                break
+            num_of_tries += 1
```

Comment:
what happens after we tried 3 times and it failed?

Reply:
![image](https://user-images.githubusercontent.com/31018228/84779969-b3fc7c80-afed-11ea-9a77-970952375419.png)